src/contracts/core/RewardsCoordinator.sol

```diff
@@ -340,6 +391,86 @@ contract RewardsCoordinator is
+            "RewardsCoordinator._validateOperatorDirectedRewardsSubmission: no strategies set"
+        );
+        require(
+            operatorDirectedRewardsSubmission.operatorRewards.length > 0,
+            "RewardsCoordinator._validateOperatorDirectedRewardsSubmission: no operators rewarded"
+        );
+
+        uint256 totalAmount = 0;
+        address currOperatorAddress = address(0);
+        for (uint256 i = 0; i < operatorDirectedRewardsSubmission.operatorRewards.length; ++i) {
+            OperatorReward calldata operatorReward = operatorDirectedRewardsSubmission.operatorRewards[i];
+            require(
+                operatorReward.operator != address(0),
+                "RewardsCoordinator._validateOperatorDirectedRewardsSubmission: operator cannot be 0 address"
+            );
+            require(
+                currOperatorAddress < operatorReward.operator,
+                "RewardsCoordinator._validateOperatorDirectedRewardsSubmission: operators must be in ascending order to handle duplicates"
+            );
+            currOperatorAddress = operatorReward.operator;
+            require(
+                operatorReward.amount > 0,
+                "RewardsCoordinator._validateOperatorDirectedRewardsSubmission: operator reward amount cannot be 0"
+            );
+            totalAmount += operatorReward.amount;
+        }
+
+        require(
+            operatorDirectedRewardsSubmission.duration <= MAX_REWARDS_DURATION,
+            "RewardsCoordinator._validateOperatorDirectedRewardsSubmission: duration exceeds MAX_REWARDS_DURATION"
+        );
+        require(
+            operatorDirectedRewardsSubmission.duration % CALCULATION_INTERVAL_SECONDS == 0,
+            "RewardsCoordinator._validateOperatorDirectedRewardsSubmission: duration must be a multiple of CALCULATION_INTERVAL_SECONDS"
+        );
+        require(
+            operatorDirectedRewardsSubmission.startTimestamp % CALCULATION_INTERVAL_SECONDS == 0,
+            "RewardsCoordinator._validateOperatorDirectedRewardsSubmission: startTimestamp must be a multiple of CALCULATION_INTERVAL_SECONDS"
+        );
+        require(
+            block.timestamp - MAX_RETROACTIVE_LENGTH <= operatorDirectedRewardsSubmission.startTimestamp &&
+                GENESIS_REWARDS_TIMESTAMP <= operatorDirectedRewardsSubmission.startTimestamp,
+            "RewardsCoordinator._validateOperatorDirectedRewardsSubmission: startTimestamp too far in the past"
+        );
+        require(
+            operatorDirectedRewardsSubmission.startTimestamp + operatorDirectedRewardsSubmission.duration <
+                block.timestamp,
+            "RewardsCoordinator._validateOperatorDirectedRewardsSubmission: operator-directed rewards submission is not retroactive"
+        );
+
+        // Require operatorDirectedRewardsSubmission is for whitelisted strategy or beaconChainETHStrategy
+        address currAddress = address(0);
```

Comment:
can we abstract any of this logic internally to be used in the other rewards functions

Reply:
Oh you mean the validations. Yup can do that.

Reply:
Done in [caf848c](https://github.com/Layr-Labs/eigenlayer-contracts/pull/837/commits/caf848cfa3a22b38d65085f6c4e48bdf23acce8e)